### PR TITLE
Random recipe card

### PR DESCRIPTION
**Pull request by hipro90**
add waiting component and add array of ingredients in recipeCard.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,12 +3,15 @@ import React from 'react';
 import Header from './components/Header'
 import Navbar from './components/Navbar'
 import Footer from './components/Footer'
-
 import Accueil from './components/Accueil'
 import FullRecipeCard from './components/FullRecipeCard'
+import RecipeSearch from './components/RecipeSearch'
+import BeerCard from './components/BeerCard'
+// import Calculator from './components/Calculator'
```
Comment (clemderome) [inline src/App.js:10]: delete this comment

File: src/components/FullRecipeCard.css [hunk 2]
```diff
@@ -55,14 +58,31 @@ button.button-recipe{
     margin: 1.5em;
 }
 
+.retour{
+    width: 110px;
```
Comment (clemderome) [inline src/components/FullRecipeCard.css:62]: transform to em , no pixels pls

File: src/components/Waiting.css [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.eyesLeft{
+    border-radius: 50%;
+    background-color: white;
+    margin-right: -82px;
+    z-index: 20;
+    margin-bottom: 61px;
+    width: 36px;
+    height: 36px;
```
Comment (clemderome) [inline src/components/Waiting.css:12]: no pixels, em pls

**Review by clemderome — APPROVED** [review]
Good, some little css to change but it's ok

File: src/components/RecipeCard.js [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
 class RecipeCard extends React.Component {
 
   state = {
     recipe: '',
     ingredients: [],
-    UserChoice: false,
-    page: ''
+    bool:false
   }
+  getRecipe(ingredient1, ingredient2, ingredient3, userCalories) {
```
Comment (jerkodeur) [inline src/components/RecipeCard.js:15]: Retirer les paramètres devenus obsolètes

File: src/components/RecipeCard.js [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
 
-  getRecipe() {
-    let selectedIngredients = "cheese"
+    let selectedIngredients = `${ingredientsArray[Math.floor(Math.random() * ingredientsArray.length)]}`
+    const allIngredient = ingredient1 && `${ingredient1},${ingredient2},${ingredient3}`
```
Comment (jerkodeur) [inline src/components/RecipeCard.js:18]: Obsolete

File: src/components/RecipeCard.js [hunk 6]
```diff
@@ -1,25 +1,29 @@
...
     const min = 0
-    const max = min + 50
-    const minCalories = 5000
-    const maxCalories = 10000
-    let url = `https://api.edamam.com/search?q=${selectedIngredients}&from=${min}&to=${max}&calories=${minCalories}-${maxCalories}&app_id=812f083c&app_key=57cd06930f1a1d5818380b512897cc58`
-
+    const max = min + 100
+    const minCalories = userCalories ? parseInt(userCalories) : 5000
```
Comment (jerkodeur) [inline src/components/RecipeCard.js:21]: A simplifier

File: src/components/RecipeCard.js [hunk 7]
```diff
@@ -1,25 +1,29 @@
...
+    const maxCalories = minCalories + 5000
+    const customIngredient = allIngredient ? allIngredient : selectedIngredients
+    console.log(selectedIngredients)
```
Comment (jerkodeur) [inline src/components/RecipeCard.js]: Retirer le console.log

File: src/components/RecipeSearch.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+  const [userCalories, setUserCalories] = useState(0)
+  const [userDiets, setUserDiets] = useState('')
+  const [errorRequest, setErrorRequest] = useState(false)
+  const [recipe, SetRecipes] = useState([])
+
+  // Generate a random number
+  const getRandomNumber = (max) => Math.floor(Math.random() * Math.floor(max))
+  //Define the range of search for the api request
+  const defineRangeNumber = (nbResults) => {
+    console.log(nbResults, "firstresult")
```
Comment (jerkodeur) [inline src/components/RecipeSearch.js]: Retirer le console.log

File: src/components/RecipeSearch.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    const rangewidth = 10
+    const numberToRandom = getRandomNumber(Math.ceil(nbResults / rangewidth))
+    console.log(numberToRandom, "numberRandom")
```
Comment (jerkodeur) [inline src/components/RecipeSearch.js]: Retirer le console.log

File: src/components/RecipeSearch.js [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    const max = (numberToRandom * rangewidth) + rangewidth > nbResults ? nbResults - 1 : (numberToRandom * rangewidth) + rangewidth
+    console.log(max, "max")
```
Comment (jerkodeur) [inline src/components/RecipeSearch.js]: Retirer le console.log

File: src/components/RecipeSearch.js [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+    const min = max - rangewidth < 0 ? 0 : max - rangewidth
+    return `&from=${min}&to=${max}`
+  }
+
+  const defineRequestUrl = (nbResults) => {
+    nbResults = nbResults > 100 ? 100 : nbResults
+    const calories = userCalories && `&calories=${userCalories}`
+    const ingredients = userIngredient1 && `${userIngredient1},${userIngredient2},${userIngredient3}`
+    const diet = userDiets && `&health=${userDiets}`
+    console.log(diet, "diet")
```
Comment (jerkodeur) [inline src/components/RecipeSearch.js]: Retirer le console.log

**Review by jerkodeur — CHANGES_REQUESTED** [review]
Just some console/log to remove and any obsolete code to remove if possible

**Review by jerkodeur — APPROVED** [review]
The console.log had been remove

**Pull request merged**